mlir/lib/Analysis/DataFlow/DenseAnalysis.cpp

```diff
@@ -139,15 +139,43 @@ void AbstractDenseDataFlowAnalysis::visitRegionBranchOperation(
   assert(predecessors->allPredecessorsKnown() &&
          "unexpected unresolved region successors");
 
+  auto getBeforeBranchState = [&](RegionBranchOpInterface branch) {
+    const AbstractDenseLattice *before;
+    if (Operation *prev = branch->getPrevNode())
+      before = getLatticeFor(point, prev);
+    else
+      before = getLatticeFor(point, branch->getBlock());
+    return before;
+  };
+
+  if (auto *op = point.dyn_cast<Operation *>()) {
+    if (op == branch)
+      // In this context, the known predecessors are the last operations along
+      // all execution paths that pass through the branch. If the number of
+      // branch regions is equal to the number of terminates (e.g., scf.yield)
+      // plus one, then we need to join the state before the branch, as the
```

Comment:
When the number of terminators (e.g. scf.yield) is less than the branch operation regions, we need to join ...